packages/remix-react/transition.ts

```diff
@@ -811,7 +811,11 @@ export function createTransitionManager(init: TransitionManagerInit) {
       update({
         fetchers: new Map(state.fetchers),
         error: result.value,
-        errorBoundaryId
+        errorBoundaryId,
+        location: {
+          ...state.location,
+          state: { isRedirect: false, type: "error" }
```

Comment:
The issue seems to be that `errorBoundaryId` was being set here, but the error boundary only updates its error state when the `location` changes. Removing that check caused other tests to fail, so I added a location change.